src/main/java/org/example/gameService/SaveGame.java

```diff
@@ -0,0 +1,65 @@
+package org.example.gameService;
+
+import org.example.model.GameData;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+
+public class SaveGame {
+
+    private VCSFileCreator vcsFileCreator;
+    private GameData gameData;
+    private LocalDate date;
+    private Path path;
+    private File file;
+    private String gameHistoryFilePath;
+    private DateTimeFormatter formatter;
+    private LocalTime time;
+
+    public SaveGame(GameData gameData) {
```

Comment:
Why does SaveGame get gameData in the constructor, when the method responsible for saving is using other gameData?

Reply:
Previously, I quckly created the VCSFileCreator class from SaveGame class, which resulted in poor considered fields and poor structured methods in both the SaveGame and VCSFileCreator classes. In the current commit, I have cleaned up the code by removing unnecessary fields and improving the methods.